finality-aleph/src/party/member.rs

```diff
@@ -23,17 +30,28 @@ pub fn task<
     network: NetworkWrapper<AlephNetworkData<B>, ADN>,
     data_provider: impl aleph_bft::DataProvider<AlephData<B>> + Send + 'static,
     ordered_data_interpreter: OrderedDataInterpreter<B, C>,
+    backup_saving_path: Option<PathBuf>,
 ) -> Task {
     let AuthoritySubtaskCommon {
         spawn_handle,
         session_id,
     } = subtask_common;
     let (stop, exit) = oneshot::channel();
-    // `sink` and `empty` here are noop placeholders which will be replaced in A0-542
-    let local_io = LocalIO::new(data_provider, ordered_data_interpreter, sink(), empty());
     let task = {
         let spawn_handle = spawn_handle.clone();
         async move {
+            let (saver, loader) = match rotate_saved_backup_files(backup_saving_path, session_id) {
```

Comment:
Hm, does this have to happen within the future? I kinda suspect it does, because file handles might not be `Send` or something, but if by any chance it could be outside it would be much nicer, and you could return a `Result` from this function, leaving the logging to the higher level.

Reply:
Most futures _are_ `Send`, so that probably won't be an issue. But is it okay to do the rotation _before_ the task is started like that?

Reply:
This rotation only touches files related to a single session, right? If so, then it definitely can be done earlier, there won't be anything else happening to this session within this program run.

If this also does some previous session cleaning then possibly no (so we probably shouldn't do it,'cause it would be bug prone), but I think this was in another part of the code, right? <_<

Reply:
Backup cleaning happens after this future is done. So this definitely does not need to happen within the future. Also both BackupSave/Loader are Send, so can be passed. I wanted to keep this function as it is without returning errors, so I moved rotation inside the future. So that all we keep it as it is (task is created always) and after that it might fail. Maybe moving it outside future would be better, but it kind of causes us to handle this error in mamy places? It would have to be returned from `member::task` after that from `spawn_authority_subtasks`, after that from `spawn_authority_task`, aaand after that handled in `run_session`. Right now it is handled anyway, as session termination prematurely is handled in `run_session`. WDYT?

Reply:
I think a fail early approach is better – if we can return an error earlier, before creating the future, we should. It previously didn't return errors, because there were no pre-future errors to return.

And the error would only be handled in one place, wherever the task creation happens.

Reply:
I see why this would be better, but backup reading could kind of be part of running a member. It also makes our code more complicated. This is the only error that might happen before creating the future and to understand it the person would have to jump several functions in the code. Right now I'm split between those two approaches. 

Just to be clear. We would add Result returning from `member::task`, `spawn_authority_subtask`, `spawn_authority_task`. After that log it in `run_session` and return `None` as `maybe_authority_task`?

Reply:
Or even propagate it further, since not being able to run a session as an authority when the node should is quite problematic.

But if nothing on this path returns or logs errors then perhaps this is a bigger task and should be created as such?

Reply:
Hmmm, but it is handled now. There are some logs in Subtasks. Still maybe it should be handled better. I would go with a bigger task and leave it as it is now. I feel like error handling here should be looked at also from other perspectives